### ISSUE TITLE
Can't derive on structs containing Vec<T>

### ISSUE THREAD

**Issue by alecmocatta** [koute/speedy#8]
This example fails:

```rust
use speedy::{Readable, Writable};

#[derive(Readable, Writable)]
struct Wrap<T>(Vec<T>);
```

with:

```text
error[E0277]: the trait bound `T: Readable<'a_, C_>` is not satisfied
 --> common/src/lib.rs:8:10
  |
8 | #[derive(Readable, Writable)]
  |          ^^^^^^^^ the trait `Readable<'a_, C_>` is not implemented for `T`
  |
  = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider further restricting type parameter `T`
  |
8 | #[derive(Readable, T: Readable<'a_, C_>, Writable)]
  |                  ^^^^^^^^^^^^^^^^^^^^^^

error[E0277]: the trait bound `T: Writable<C_>` is not satisfied
 --> common/src/lib.rs:8:20
  |
8 | #[derive(Readable, Writable)]
  |                    ^^^^^^^^ the trait `Writable<C_>` is not implemented for `T`
  |
  = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider further restricting type parameter `T`
  |
8 | #[derive(Readable, Writable, T: Writable<C_>)]
  |                            ^^^^^^^^^^^^^^^^^
```

The expanded code looks fine, I think this is a Rust issue. https://github.com/rust-lang/rust/issues/20671 and https://github.com/rust-lang/rust/issues/44491 look relevant.

I've worked around this by manually impling these traits. A nicer solution might be enabling a `bound` attribute, akin to [serde's](https://serde.rs/attr-bound.html)?

**Comment by koute** [comment]
Thanks. Should be fixed in 0.7.1.

**Comment by alecmocatta** [comment]
Confirmed. Thanks for the useful project and swift fix @koute!
